helix-view/src/editor.rs

```diff
@@ -581,7 +602,19 @@ impl Editor {
         let scopes = theme.scopes();
         self.syn_loader.set_scopes(scopes.to_vec());
 
-        self.theme = theme;
+        if preview {
+            if self.last_theme.is_none() {
+                // On the first preview
+                self.last_theme = Some(std::mem::replace(&mut self.theme, theme));
+            } else {
+                // Subsequent previews
+                self.theme = theme;
+            }
```

Comment:
```suggestion
            // only insert on first preview
            self.last_theme
                .get_or_insert(std::mem::replace(&mut self.theme, theme));
```
This is shorter but seemed a bit harder to understand. I am fine with either.

Reply:
TIL, reminds me of [??=](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_nullish_assignment) in other languages.

I think the confusing part is that we want to emphasize the side effect of replace happens no matter what, which can be hard to see there. Alternatively, we can switch it to this:

```suggestion
            let last_theme = std::mem::replace(&mut self.theme, theme);
            // only insert on first preview: this will be the last theme the user has saved
            self.last_theme.get_or_insert(last_theme);
```